server/sonar-server/src/test/java/org/sonar/server/usergroups/ws/UserGroupsWsTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.usergroups.ws;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.WsTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class UserGroupsWsTest {
+  @Rule
+  public UserSessionRule userSessionRule = UserSessionRule.standalone();
+  WebService.Controller controller;
+
+  @Before
+  public void setUp() {
+    WsTester tester = new WsTester(new UserGroupsWs(new SearchAction(mock(DbClient.class))));
+    controller = tester.controller("api/usergroups");
+  }
+
+  @Test
+  public void define_controller() {
+    assertThat(controller).isNotNull();
+    assertThat(controller.description()).isNotEmpty();
+    assertThat(controller.since()).isEqualTo("5.2");
+    assertThat(controller.actions()).hasSize(1);
+  }
+
+  @Test
+  public void define_search_action() {
+    WebService.Action action = controller.action("search");
+    assertThat(action).isNotNull();
+    assertThat(action.isPost()).isFalse();
```

Comment:
uneeded in my opinion. A post request will work, it's the other way around. We need to check when isPost is true